de.dlr.sc.virsat.swtbot.test/src/de/dlr/sc/virsat/swtbot/test/FuncElectricalDiagramTest.java

```diff
@@ -0,0 +1,203 @@
+
+import org.eclipse.swtbot.eclipse.gef.finder.widgets.SWTBotGefConnectionEditPart;
+import org.eclipse.swtbot.eclipse.gef.finder.widgets.SWTBotGefEditPart;
+import org.eclipse.swtbot.eclipse.gef.finder.widgets.SWTBotGefEditor;
+import org.eclipse.swtbot.swt.finder.widgets.SWTBotTreeItem;
+import org.junit.Before;
+import org.junit.Test;
+
+import de.dlr.sc.virsat.concept.unittest.util.ConceptXmiLoader;
+import de.dlr.sc.virsat.model.dvlm.concepts.Concept;
+import de.dlr.sc.virsat.model.extension.funcelectrical.model.Interface;
+import de.dlr.sc.virsat.model.extension.funcelectrical.model.InterfaceEnd;
+import de.dlr.sc.virsat.model.extension.funcelectrical.model.InterfaceType;
+import de.dlr.sc.virsat.model.extension.funcelectrical.model.InterfaceTypeCollection;
+import de.dlr.sc.virsat.model.extension.ps.model.ConfigurationTree;
+import de.dlr.sc.virsat.model.extension.ps.model.ElementConfiguration;
+
+
+
+/**
+ * 
+ * SWTBot test class for testing Funcelectrical Diagrams
+ *
+ */
+public class FuncElectricalDiagramTest extends ASwtBotTestCase {
+	
+	private SWTBotTreeItem repositoryNavigatorItem;
+	private SWTBotTreeItem configurationTree;
+	private SWTBotTreeItem elementConfiguration;
+	private SWTBotGefEditor diagramEditor;
+	
+	private Concept conceptFuncElectrical;
+	
+	private static final int DRAG1_COORDINATES_X = 100;
+	private static final int DRAG1_COORDINATES_Y = 100;
+	
+	private static final int DRAG2_COORDINATES_X = 100;
+	private static final int DRAG2_COORDINATES_Y = 200;
+	
+	private static final String ELEMENT_CONFIGURATION_NAME = "ElementConfiguration";
+
+	@Before
+	public void before() throws Exception {
+		super.before();
+		// create the necessary items for the test
+		conceptFuncElectrical = ConceptXmiLoader.loadConceptFromPlugin(de.dlr.sc.virsat.model.extension.funcelectrical.Activator.getPluginId() + "/concept/concept.xmi");
+		repositoryNavigatorItem = bot.tree().expandNode(SWTBOT_TEST_PROJECTNAME, "Repository");
+		configurationTree = addElement(ConfigurationTree.class, conceptPs, repositoryNavigatorItem);
+		elementConfiguration = addElement(ElementConfiguration.class, conceptPs, configurationTree);
+				
+		createNewDiagramForTreeItem(configurationTree, DiagramType.interfaces);
+		diagramEditor = getOpenedDiagramEditorbyTitle("newDiagram");
+	}
+	
+	@Test
+	public void addInterfaceDiagramElementUndoRedoTest() {
+		addElement(Interface.class, conceptFuncElectrical, elementConfiguration);
+		dragTreeItemToDiagramEditor(elementConfiguration, diagramEditor);
+		assertTrue(isEditPartPresentInDiagramEditor(diagramEditor, ELEMENT_CONFIGURATION_NAME));
+		undo();
+		assertFalse(isEditPartPresentInDiagramEditor(diagramEditor, ELEMENT_CONFIGURATION_NAME));
+		redo();
+		assertTrue(isEditPartPresentInDiagramEditor(diagramEditor, ELEMENT_CONFIGURATION_NAME));
+	}	
+
+	@Test
+	public void deleteObjectOutsideDiagramUpdateDiagramTest() {
+		addElement(Interface.class, conceptFuncElectrical, elementConfiguration);
```

Comment:
A little bit of textual explanation would be great. I think these test cases are not clear to everyone, in particular not for people not developing grafiti diagrams in detail.